types.go

```diff
@@ -23,12 +25,40 @@ const uuid_length = 16
 type UUID [uuid_length]byte
 
 func (u *UUID) Scan(v any) error {
-	if n := copy(u[:], v.([]byte)); n != uuid_length {
-		return fmt.Errorf("invalid UUID length: %d", n)
+	switch val := v.(type) {
+	case []byte:
+		if len(val) != uuid_length {
+			return u.Scan(string(val))
+		}
+		copy(u[:], val[:])
+	case string:
+		id, err := uuid.Parse(val)
+		if err != nil {
+			return err
+		}
+		copy(u[:], id[:])
+	default:
+		return fmt.Errorf("invalid UUID type: %#v", val)
 	}
```

Comment:
Thanks for the changes! Do we already have tests covering these (error) code paths?

Reply:
Not yet, but I can add one test case to do that.

Reply:
Perfect! :)

Reply:
All three error paths has been tested (type check, []byte values check and string values check). You could take a look.